Source/WarpX.H

```diff
@@ -116,6 +116,7 @@ public:
     void InitData ();
 
     void Evolve (int numsteps = -1);
+    void Synchronize ();
```

Comment:
Can you please add a doc-string for what this function does (we're trying to improve the WarpX inline commenting to make things easier to understand 😃 )